### PR TITLE
chore: use long IDs for data approval [DHIS2-11370]

### PR DESCRIPTION
**Pull request by jbee**
Changes PK ids from `int` to `long` for `DataApproval` and `DataApprovalAudit`.
Also cleanup of their `equals` implementation using `Objects.equals` instead of if-else construct.

### CODE REVIEW

**Review by jimgrace — APPROVED** [review]
Huh. Somebody went through and upgraded IDs from int to long some time ago. I hadn't noticed that they had missed data approvals. Also thanks for the equals modernization! The code looks good; I can't find any usage of the IDs beyond what you have changed.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApproval.java [hunk 1]
```diff
@@ -336,74 +337,17 @@ public boolean equals( Object object )
...
-            return false;
-        }
+        DataApproval other = (DataApproval) object;
 
-        return true;
+        return Objects.equals( dataApprovalLevel, other.dataApprovalLevel )
+            && Objects.equals( workflow, other.workflow )
+            && Objects.equals( period, other.period )
+            && Objects.equals( organisationUnit, other.organisationUnit )
+            && Objects.equals( attributeOptionCombo, other.attributeOptionCombo );
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApproval.java:351]: Looks good. Should we use `Objects.hash` or `Objects.hashCode` for `hashCode` just to make it consistent and equally modern while at it?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApprovalAudit.java [hunk 2]
```diff
@@ -275,114 +277,21 @@ public boolean equals( Object object )
...
-            if ( !creator.equals( that.getCreator() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getCreator() != null )
+        if ( !(object instanceof DataApprovalAudit) )
         {
             return false;
         }
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApprovalAudit.java:284]: Looks good. Should we use Objects.hash or Objects.hashCode for hashCode just to make it consistent and equally modern while at it?

**Review by larshelge — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApproval.java (hunk 1 above)
Comment (jbee) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApproval.java:341]: Definite proof it was too warm lately: I totally forgot `Objects.hash` existed :grin:

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApprovalAudit.java (hunk 2 above)
Comment (jbee) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApprovalAudit.java:270]: We absolutely should. Done.

**Pull request merged**